armv7a9-zynq7000/_startup.S

```diff
@@ -0,0 +1,233 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * Low-level initialization for Cortex-A9 (ARMv7) architecture
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#define __ASSEMBLY__
+
+#include "cpu.h"
+#include "peripherals.h"
+
+.arm
+
+.section .init, "ax"
+.globl _vector_table
+
+
+.global _start
+.type _start, %function
+
+
+/* Vector Table Definition */
+.org 0
+_vector_table:
+	b _start
+	b _exception_undef
+	b _syscalls_dispatch
+	b _exception_prefetch
+	b _exception_abort
+	.word 0
+	b _interrupts_dispatch
+	b _interrupts_dispatch
+
+
+/* startup code */
+_start:
+	cpsid aif, #SYS_MODE
+
+	/* Only CPU0 performs initialization, others go into wfi */
+	mrc p15, 0, r1, c0, c0, 5       /* Read Multiprocessor Affinity Register */
+	and r1, r1, #0xf                /* Extract CPU ID                        */
+	cmp r1, #0
+	beq initialize
+
+/* TODO: make appropriate action when other core than CPU_0 is running */
+wait_loop:
+	wfi
+	b wait_loop
```

Comment:
According to documentation:
_CPU 0 is in charge of starting code execution on CPU 1. The BootROM puts CPU 1 into the Wait for
Event mode. Nothing has been enabled and only a few general purpose registers have been
modified to place it in a state where it is waiting at the WFE instruction._

I've added this instruction to prevent undefined behaviour in case of running this code on the other core.

Reply:
So according to the documentation the other core shouln't be running on boot, we need to send an event to it. We may leave this code for now, just to be double sure that it isn't in fact running.